ui/src/views/user/survey-edit/SurveyEditView.vue

```diff
@@ -240,22 +249,60 @@ export default {
       return this.formData;
     },
     onSaveAsDraftOptionClick() {
+      console.log("save as draft");
       const finalOutput = this.getData();
       console.log(JSON.stringify(finalOutput));
 
-      this.$notify.toast("Survey has been successfully saved");
+      if(finalOutput.config.startDate == "" || finalOutput.config.endDate == "") {
+        this.$notify.toast("Please give survey's start date and end date!");
+      } else {
+        if (this.currentSurveyId == "new") {
+          userAddSurvey(finalOutput)
+          .then(() => {
+            this.$notify.toast("Your survey has been saved!");
+            this.$router.push("/user/surveys/");
+          }).catch((error) => {
+            this.$notify.toast(error["message"]);
```

Comment:
This is for future reference, but I see that in your PRs most of object property access is done by [], even though we can also do it with the dot operator ".". [] is preferred when the key is dynamic, but if its static, better to use "." instead.

Reply:
thanks for the hint 😅 well, I noticed that my code has some inconsistencies because recently I accesed object property with dot. I will refactor it to [""], for the sake of consistency with the older code

In the future, I will keep that in mind to use dot for static keys. Thanks!